desc/objectives/getters.py

```diff
@@ -132,20 +120,13 @@ def get_fixed_boundary_constraints(eq, profiles=True, normalize=True):
         A list of the linear constraints used in fixed-boundary problems.
 
     """
-    constraints = (
-        FixBoundaryR(eq=eq, normalize=normalize, normalize_target=normalize),
-        FixBoundaryZ(eq=eq, normalize=normalize, normalize_target=normalize),
-        FixPsi(eq=eq, normalize=normalize, normalize_target=normalize),
-    )
+    kwargs = {"eq": eq, "normalize": normalize, "normalize_target": normalize}
+    constraints = (FixBoundaryR(**kwargs), FixBoundaryZ(**kwargs), FixPsi(**kwargs))
     if profiles:
         for name, con in _PROFILE_CONSTRAINTS.items():
             if getattr(eq, name) is not None:
-                constraints += (
-                    con(eq=eq, normalize=normalize, normalize_target=normalize),
-                )
-    for param in ["I", "G", "Phi_mn"]:
-        if np.array(getattr(eq, param, [])).size:
-            constraints += (FixParameter(eq, param),)
+                constraints += (con(**kwargs),)
+    constraints += (FixSheetCurrent(**kwargs),)
```

Comment:
does this add the sheet current constraint even when it might not have a sheet current?

Reply:
Technically yes, it always adds the sheet current constraints. But the equilibrium always has those optimizable parameters, they are just often set to empty arrays. In that case these constraints are size 0 so there is no harm done.